### PR TITLE
CFE-740: Updated cert-manager-operator API to enable customization

### PR DESCRIPTION
**Pull request by thejasn**
## Description

Update operator API to support customization of operand deployments (controller, webhook, cainjector). This reduces end user dependency on using `UnsupportedConfigOverrides` which is not supported as the name indicates.

### CODE REVIEW

File: bundle/manifests/operator.openshift.io_certmanagers.yaml [hunk 1]
```diff
@@ -34,6 +34,263 @@ spec:
...
+                      type: string
+                    type: array
+                  overrideEnv:
+                    items:
+                      description: EnvVar represents an environment variable present
+                        in a Container.
+                      properties:
+                        name:
+                          description: Name of the environment variable. Must be a
+                            C_IDENTIFIER.
```
Comment (TrilokGeer) [inline bundle/manifests/operator.openshift.io_certmanagers.yaml:57]: Is is possible to have regular expression pattern?

Comment (thejasn) [reply bundle/manifests/operator.openshift.io_certmanagers.yaml:57]: Yes, kubebuilder does provide regex pattern. `+kubebuilder:validation:Pattern:
string` . ([ref](https://book.kubebuilder.io/reference/markers/crd-validation.html))

File: config/crd/bases/operator.openshift.io_certmanagers.yaml [hunk 2]
```diff
@@ -35,6 +35,263 @@ spec:
           spec:
             description: CertManagerSpec defines the desired state of CertManager
             properties:
+              cainjectorConfig:
+                description: "CAInjectorConfig specifies further customization options
+                  for the cainjector's deployment spec. \n Possible customizations
+                  include the following, \n For OverrideArgs, --namespace string                          If
+                  set, this limits the scope of cainjector to a single namespace.
+                  If set, cainjector will not update resources with certificates outside
+                  of the configured namespace. --v Level                                   number
```
Comment (TrilokGeer) [inline config/crd/bases/operator.openshift.io_certmanagers.yaml:44]: nit :extra spaces

**Comment by thejasn** [comment]
/cc @xingxingxia 
This will be followed up with #95, so most of the testing would need to be done as part of that.

**Comment by xingxingxia** [comment]
Intended to review/research to do a pre-merge test before labeling. Due to no enough time currently, let me first add it:
/label qe-approved
Will test after merge.

**Pull request merged**